contracts/UnionZap.sol

```diff
@@ -49,22 +53,33 @@ contract UnionZap is Ownable, UnionBase {
         bytes32[] merkleProof;
     }
 
+    struct tokenContracts {
+        address pool;
+        address swapper;
+        address distributor;
+    }
+
     struct curveSwapParams {
         address pool;
         uint16 ethIndex;
     }
 
+    mapping(address => tokenContracts) public tokenInfo;
     mapping(address => curveSwapParams) public curveRegistry;
 
     event Received(address sender, uint256 amount);
-    event Distributed(uint256 amount, uint256 fees, bool locked);
-    event DistributorUpdated(address distributor);
+    event Distributed(uint256 amount, address token, address distributor);
     event VotiumDistributorUpdated(address distributor);
     event FundsRetrieved(address token, address to, uint256 amount);
     event CurvePoolUpdated(address token, address pool);
+    event OutputTokenUpdated(
+        address token,
+        address pool,
+        address swapper,
+        address distributor
+    );
 
     constructor(address distributor_) {
-        unionDistributor = distributor_;
         routers[0] = SUSHI_ROUTER;
```

Comment:
Since the `SUSHI_ROUTER` and `UNISWAP_ROUTER` variables aren't used anywhere else outside of this constructor, it'd make sense to remove them and use the addresses directly, or have them passed to the constructor.

Reply:
I don't think using the addresses directly would create any gas savings (for the compiler it'd still be a constant?) so it would just make the code less readable?

Reply:
Imo having less contract variables (or code, if possible) would help with code reviewal or audits - but if you feel it helps then feel free to keep them ser